woocommerce/woocommerce-paypal-payments
# Unblock UI if Credit Card payment failed

**Issue**: [PCP-66](https://inpsyde.atlassian.net/browse/PCP-66).

---

### Description

Failed `validate-payment-method` request causes the plugin to hang. The UI is blocked forever, the customer can see a spinner that never goes away.

### Steps to test:


1. Enable sandbox mode in the plugin settings.
2. Block outgoing requests to `cors.api.sandbox.paypal.com` to make the request fail.
    Adding to your `hosts` file (`/etc/hosts` for Linux-based OS) a line like `127.0.0.1 cors.api.sandbox.paypal.com` should be enough.
3. Add any product to your cart, proceed to checkout and fill the form.
4. Choose `Credit Cards` as a payment method, enter any data for the card number, CVV, and date.
5. Open your browser console.
6. Submit checkout.
7. You should see a failed request to the `cors.api.sandbox.paypal.com`, then your UI should be unblocked and you should see a standard Woocommerce order notice with the message `Something went wrong. Please try again or choose another payment source.`.
8. Don't forget to remove the line you added to the `hosts` file :wink: 

### Documentation




### Changelog entry
* Fix - Never unblocking UI in case of an error on checkout when using a credit card.

### Note
The GitHub Actions failed because of a composer error, the same as for the parent `trunk` branch. This error is not related to changes from this PR.
